src/allmydata/util/eliotutil.py

```diff
@@ -320,7 +324,8 @@ def decorate_log_call_deferred(f):
         def logged_f(*a, **kw):
             # Use the action's context method to avoid ending the action when
             # the `with` block ends.
-            with start_action(action_type=action_type).context():
+            args = {k: bytes_to_unicode(True, kw[k]) for k in kw}
+            with start_action(action_type=action_type, **args).context():
```

Comment:
I can't imagine where a test would fit in here currently because:

- `bytes_to_unicode` is  a tested function, required to help covert decorated function parameters containing bytes to JSON.  (The function was introduced for python 3)
-  `log_call_deferred` would decorate parameterized functions successfully, but their arguments would not be logged without passing `**args` to start_action, which also is tested in eliot.

All that happens here is that, we pass the keyword arguments from the wrapper function to start_action so they can be logged.

That said, if we needed to introduce  a couple of unit tests what would be we be testing?

Reply:
> `log_call_deferred` would decorate parameterized functions successfully, but their arguments would not be logged without passing `**args` to `start_action`, which also is tested in eliot.

Where is this tested?  This sounds like just the piece that is not tested: that the arguments are logged because `**args` is being passed to `start_action` now.

Ideally this would be tested alongside the other unit tests for `log_call_deferred` which are in `src/allmydata/test/test_eliotutil.py`, in the `LogCallDeferredTests` test case.

If they exist elsewhere, maybe we can move them (or if there is some interesting reason not to move them, that's probably okay too - but maybe we can make them more discoverable somehow).

If they don't exist elsewhere, then adding them to `LogCallDeferredTests` would be great.  I expect they would look like tests that use the `log_call_deferred` decorator and then make assertions about the dictionaries that are delivered to a log observer, perhaps with the help of the `assertHasAction` helper (which can make assertions about the fields in the start message for an action).

Reply:
Yes that's possible. It's now clear what should be tested. My last argument was titled towards the argument that `start_action` is tested with and without `**args` in `eliot` lib itself not in Tahoe.  So I wanted to be double sure if you want those tests written over here too.